packages/next/build/index.ts

```diff
@@ -780,44 +786,80 @@ export default async function build(
               )
             }
 
-            // Build client first
-            clientResult = await runCompiler(clientConfig, {
-              runWebpackSpan,
-            })
-
-            // Only continue if there were no errors
-            if (!clientResult.errors.length) {
-              serverResult = await runCompiler(configs[1], {
+            if (useParallelBuilds) {
+              [clientResult, serverResult, edgeServerResult] = await Promise.all(
+                [
+                  runCompiler(configs[0], {
+                    runWebpackSpan,
+                  }),
+                  runCompiler(configs[1], {
+                    runWebpackSpan,
+                  }),
+                  configs[2]
+                ? runCompiler(configs[2], { runWebpackSpan })
+                : null
+                ]
+              )
+            } else {
+              // Build client first
+              clientResult = await runCompiler(clientConfig, {
                 runWebpackSpan,
               })
-              edgeServerResult = configs[2]
-                ? await runCompiler(configs[2], { runWebpackSpan })
-                : null
+
+              // Only continue if there were no errors
+              if (!clientResult.errors.length) {
+                serverResult = await runCompiler(configs[1], {
+                  runWebpackSpan,
+                })
+                edgeServerResult = configs[2]
+                  ? await runCompiler(configs[2], { runWebpackSpan })
+                  : null
+              }
             }
           } else {
             // During the server compilations, entries of client components will be
             // injected to this set and then will be consumed by the client compiler.
             injectedClientEntries.clear()
 
-            serverResult = await runCompiler(configs[1], {
-              runWebpackSpan,
-            })
-            edgeServerResult = configs[2]
-              ? await runCompiler(configs[2], { runWebpackSpan })
-              : null
-
-            // Only continue if there were no errors
-            if (
-              !serverResult.errors.length &&
-              !edgeServerResult?.errors.length
-            ) {
+            const setClientEntries = () => {
               injectedClientEntries.forEach((value, key) => {
                 ;(clientConfig.entry as webpack.EntryObject)[key] = value
               })
+            }
 
-              clientResult = await runCompiler(clientConfig, {
+            if (useParallelBuilds) {
+              setClientEntries();
+
+              [serverResult, edgeServerResult, clientResult] = await Promise.all([
```

Comment:
Is it possible to add support for limiting the parallel counts? E.g., allow `experimental.parallelBuilds` to be set to `2` and `3`.

You can import [`p-limit`](https://www.npmjs.com/package/p-limit) library from `next/dist/compiled/p-limit`:

```js
import pLimit from 'next/dist/compiled/p-limit'
```